### DISCUSSION TITLE
Process delayed jobs before they reach available_at

### DISCUSSION THREAD

**Discussion by foremtehan** [laravel/framework#35477] [Ideas]
Currently there are no way process delayed job (before they reach the `available_at`, It would be nice if we could process all delayed job at the time with artisan https://stackoverflow.com/questions/65129474/how-to-process-delayed-jobs-before-they-reach-available-at

@themsaid

**Comment by foremtehan** [comment]
@paras-malhotra [you mentiond ](https://github.com/laravel/framework/pull/35472#issuecomment-738027213) "only queueing feature that was missing in Laravel" i think there is another one is this thread

**Comment by paras-malhotra** [comment]
I don't think it's a very common use case to process it before the delay is reached. None of the backend frameworks support this and it can be cumbersome to support with drivers such as Redis imo.

**Reply by foremtehan** [reply]
The backend frameworks dont support this feature because they dont catch the idea IMHO

**Reply by paras-malhotra** [reply]
What would be a use case for delaying a job for say 10 minutes and then trying to process it early at the 5 minute mark?

**Comment by themsaid** [comment]
Can you share the use case?

**Comment by foremtehan** [comment]
Ok my use case:

I crawls bunch of news websites and share theirs articles in my website, Some of them have more priority to the others say washingtonpost, So in my app i determined if the current article come from washingtonpost set a key in cache (db driver) and dispatch it otherwise delay any other for about 1 hour:

```php
if (Indexer::is() == 'washingtonpost') {
    SendArticle::disptach(Indexer::link());

    Cache::put('locked', true, now()->addHour());
} else {
    if (Cache::has('locked')) {
        SendArticle::dispatch(Indexer::link())->delay(now()->addHour());
    } else {
        SendArticle::dispatch(Indexer::link());
    }
}
```
And now imagine that all of the articles that comes after the washingtonpost are delayed at different times. For example, an article may be delayed for 1 hour after 10 minutes , etc...

Say 10 article delayed in diffrenet times and ready to publish But the problem here is that whenever the key (locked) is deleted, not all articles are sent immediately because they are delayed at different times So i would really appreciate if there was something that checks if we have delayed jobs ? then dispatch all of them immediately! and dont wait until you reach `available_at` Something like:

```php

if(DelayedJobs::exists())
{
    DelayedJobs::disptachAll();
} 
```

Thanks for your responses guys much appreciated!

**Reply by themsaid** [reply]
Seems like what you need is multiple queues with priorities.

**Reply by foremtehan** [reply]
So does it worth to implement? 😅
